src/simplnx/Utilities/ThreadedAlgorithm.hpp

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+#include "simplnx/Filter/IFilter.hpp"
+#include "simplnx/simplnx_export.hpp"
+
+#include <chrono>
+#include <cstdint>
+
+namespace nx::core
+{
+
+class SIMPLNX_EXPORT ThreadedAlgorithm
```

Comment:
Integrating the time counts into this is a really bad idea. I think the threads calling this should be responsible for determining when to update. The check for 1000 milliseconds is an expensive one because of all the casts and it must be locked for that entire duration. This will significantly slow down output, rather than just locking to calculate and assemble the string output. By outsourcing the time cast checks to the threads at least we leverage the parallel nature. That said I will have to verify that the chrono library is actually threadsafe.

Edit: steady_clock is threadsafe: https://stackoverflow.com/questions/10894504/is-there-any-stdchrono-thread-safety-guarantee-even-with-multicore-context

Reply:
This implementation is the same as what some filters were already doing. I'd say that for this PR getting more filters to output and unifying the method of output is the goal. Then it should be easier to update the implementation to something with improved performance later.